### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by simondeziel**

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ jobs:
           TARGET: ${{ github.ref_name }}
         run: |
           set -x
+          git config --global transfer.fsckobjects true
```
Comment (tomponline) [inline .github/workflows/builds.yml:44]: What does this do/prevent?

Comment (simondeziel) [reply .github/workflows/builds.yml:44]: This is a setting that can catch malformed objects and prevent/mitigate certain attacks. It was mentioned as a mitigation to some CVEs (like https://securityaffairs.com/73051/hacking/cve-2018-11235-git-flaw.html). https://resources.infosecinstitute.com/topics/security-awareness/security-best-practices-for-git-users/ recommends setting it (practice 19) so does https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=813157. FYI, setting `transfer.fsckobjects` also provides default values for `fetch.fsckObjects` and `receive.fsckObjects`.

**Review by tomponline — APPROVED** [review]

File: .github/workflows/builds.yml (hunk 1 above)
Comment (tomponline) [reply .github/workflows/builds.yml:44]: @simondeziel thanks could you also apply this change to the lxd, microcloud and microcloud-pkg-snap repos?

Comment (simondeziel) [reply .github/workflows/builds.yml:44]: Done:
https://github.com/canonical/lxd/pull/12342
https://github.com/canonical/microcloud/pull/188
https://github.com/canonical/microcloud-pkg-snap/pull/4

**Pull request merged**